File: challenges/66/solution.py
def plusOne(digits):
    carry = 1
    i = len(digits)-1
    while(i >= 0):
        digits[i] = (digits[i] + carry) % 10

        carry = 1 if digits[i] == 0 else 0

        if(not carry):
            # no carry, we're done
            return digits
        i -= 1
    
    # insert leading 1 if carry is still 1
    digits.insert(0, 1)
    return digits

File: challenges/66/solution.ts
function plusOne(digits: number[]): number[] {
  let curry = 1;
  for (let j = digits.length - 1; j >= 0; j--) {
    digits[j] = (digits[j] + curry) % 10;
    digits[j] === 0 ? (curry = 1) : (curry = 0);
    if (!curry) return digits;
  }
  if (curry === 1) digits.unshift(curry);
  return digits;
}
